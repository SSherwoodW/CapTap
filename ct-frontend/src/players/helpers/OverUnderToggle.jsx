import { useState } from 'react';
import { RadioGroup } from '@headlessui/react';


function OverUnderToggle({ selectedValue, onChange }) {
    

  return (
    <div className="py-2 px-4">
      <RadioGroup value={selectedValue} onChange={onChange}>
        <div className="">
          {['Over', 'Under'].map((option) => (
            <RadioGroup.Option
              key={option}
              value={option}
              className={({ active, checked }) =>
                `${
                  active
                    ? 'ring-2 ring-white/60 ring-offset-2 ring-offset-sky-300'
                    : ''
                }
                ${checked ? 'bg-green text-white ring-1 ring-white/60' : 'bg-gray-500 text-white'}
                relative flex cursor-pointer text-center rounded-sm px-5 py-2 mb-2 shadow-md focus:outline-none`
              }
            >
              {({ active, checked }) => (
                <div className="flex items-center justify-center">
                  <RadioGroup.Label
                    as="p"
                    className={`font-medium ${
                      checked ? 'text-gray-100' : 'text-gray-100'
                    }`}
                  >
                    {option}
                  </RadioGroup.Label>
                  <input
                    type="hidden"
                    value={option}
                  />
                </div>
              )}
            </RadioGroup.Option>
          ))}
        </div>
      </RadioGroup>
    </div>
  )
};


export default OverUnderToggle;
