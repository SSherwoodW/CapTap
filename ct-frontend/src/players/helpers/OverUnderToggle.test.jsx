import { RadioGroup } from '@headlessui/react';
import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, act } from '@testing-library/react';

import OverUnderToggle from './OverUnderToggle';

describe('OverUnderToggle', () => {
    let container;

    beforeEach(() => {
        // Set up a container for rendering the component
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        // Clean up after each test
        document.body.removeChild(container);
        container = null;
    });

    it('renders the component with default props', () => {
        render(<OverUnderToggle/>, { container });
    
        expect(container.innerHTML).toMatchSnapshot();
    });

    it('renders the component with selectedValue correctly', () => {
        const selectedValue = 'Over';

        render(<OverUnderToggle/>, { container, props: { selectedValue } });

        // Check if the correct option is selected
        expect(container.innerHTML).toMatchSnapshot();
    });

    it('calls onChange when an option is selected', () => {
        // Mock the onChange function
        const onChange = vi.fn();

        render(<OverUnderToggle onChange={onChange}/>, { container});
        screen.debug();
        // Find the radio group
        const overButton = screen.getByText('Over')
        act(() => {
            overButton.click();
        }); 

        expect(onChange).toHaveBeenCalledWith('Over');

        const underButton = screen.getByText('Under')
        act(() => {
            underButton.click();
        }); 

        expect(onChange).toHaveBeenCalledWith('Under');
    });
});
