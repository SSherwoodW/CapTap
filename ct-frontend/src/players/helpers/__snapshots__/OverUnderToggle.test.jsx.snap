// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OverUnderToggle > renders the component with default props 1`] = `
"<div class="mt-2 px-4"><div id="headlessui-radiogroup-:r0:" role="radiogroup"><div class="" role="none"><div class="
                bg-gray-500 text-white
                relative flex cursor-pointer text-center rounded-sm px-5 py-2 mb-2 shadow-md focus:outline-none" id="headlessui-radiogroup-option-:r1:" role="radio" aria-checked="false" tabindex="0" data-headlessui-state="" aria-labelledby="headlessui-label-:r2:"><div class="flex items-center justify-center"><p class="font-medium text-gray-100" id="headlessui-label-:r2:">Over</p><input type="hidden" value="Over"></div></div><div class="
                bg-gray-500 text-white
                relative flex cursor-pointer text-center rounded-sm px-5 py-2 mb-2 shadow-md focus:outline-none" id="headlessui-radiogroup-option-:r3:" role="radio" aria-checked="false" tabindex="-1" data-headlessui-state="" aria-labelledby="headlessui-label-:r4:"><div class="flex items-center justify-center"><p class="font-medium text-gray-100" id="headlessui-label-:r4:">Under</p><input type="hidden" value="Under"></div></div></div></div></div>"
`;

exports[`OverUnderToggle > renders the component with selectedValue correctly 1`] = `
"<div class="mt-2 px-4"><div id="headlessui-radiogroup-:r5:" role="radiogroup"><div class="" role="none"><div class="
                bg-gray-500 text-white
                relative flex cursor-pointer text-center rounded-sm px-5 py-2 mb-2 shadow-md focus:outline-none" id="headlessui-radiogroup-option-:r6:" role="radio" aria-checked="false" tabindex="0" data-headlessui-state="" aria-labelledby="headlessui-label-:r7:"><div class="flex items-center justify-center"><p class="font-medium text-gray-100" id="headlessui-label-:r7:">Over</p><input type="hidden" value="Over"></div></div><div class="
                bg-gray-500 text-white
                relative flex cursor-pointer text-center rounded-sm px-5 py-2 mb-2 shadow-md focus:outline-none" id="headlessui-radiogroup-option-:r8:" role="radio" aria-checked="false" tabindex="-1" data-headlessui-state="" aria-labelledby="headlessui-label-:r9:"><div class="flex items-center justify-center"><p class="font-medium text-gray-100" id="headlessui-label-:r9:">Under</p><input type="hidden" value="Under"></div></div></div></div></div>"
`;
